src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMParams.java

```diff
@@ -0,0 +1,1179 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class represents the hyper-parameters and parameters for {@link CoverageModelEMAlgorithm}
+ *
+ * Works both as an argument collection as well as a getter/setter.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelEMParams {
```

Comment:
I got rid of setters in the post-PR commit (except for one, which I got rid of it now as well). I also renamed it to`CoverageModelArgumentCollection`.